model/activist.go

```diff
@@ -1155,7 +1172,23 @@ func UpdateActivistData(db *sqlx.DB, activist ActivistExtra, userEmail string) (
 				fmt.Println("ERROR updating activist on mailing list:", err.Error())
 			}
 		}
-		fmt.Println("Updated: ", activist.Location, activist.City, activist.State)
+		if activistAddressChanged && activist.StreetAddress != "" && activist.City != "" && activist.State != "" {
```

Comment:
please split the body of this 'if' into a separate function (e.g. called `GetCoordinatesFromAddress`) in a separate file (single responsibility principle). i suggest the function take a single string parameter for the address and return the location and any error.

additionally, but optionally, the 'if' itself could be its own function called `updateActivistCoordinates` which would allow early returns in case of error rather than nesting the happy path inside of an `if` block.